ui/layout.js

```diff
@@ -65,6 +65,10 @@ class OverviewClone extends St.BoxLayout {
         });
         box.add_actor(searchEntryClone);
 
+        // HACK: invasively find the overview signal id that AppDisplay
+        // will use when connecting to the 'hidden' signal
+        this._overviewHiddenId = Main.overview._nextConnectionId;
```

Comment:
I don't get the mechanism here. Grepping for  `_nextConnectionId` on gnome-shell or the desktop extensions yields no results, so not sure what is setting it?

Reply:
You won't find it in neither codebases. You're looking for https://gitlab.gnome.org/GNOME/gjs/-/blob/master/modules/core/_signals.js#L41

Reply:
Ouch, this is really a hack - wouldnt it be better to push a patch upstream to fix AppDisplay to properly disconnect this signal on destruction? I dont get the issue upstream has with fixing lifecycle issues, this should be good practice.

A simple:
```
            this.connect('destroy', () => {
                Main.overview.disconnect(id);
            });
```

Right below connecting to the hidden signal would fix it (and is already done on AppDisplay when connecting to the overview hiding signal).

Reply:
Given we are already fixing other issues upstream (e.g. https://gitlab.gnome.org/GNOME/gnome-shell/-/merge_requests/1406) I'd say lets do the same here.

Reply:
Going with this as is to unblock this but please consider sending a patch upstream and removing this workaround.